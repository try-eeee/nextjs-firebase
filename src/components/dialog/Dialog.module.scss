.dialog {
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  overflow: hidden;
  border: 0;
  border-radius: 10px;
  box-shadow: 0 4px 12px rgba(0, 0, 0, 0.1);

  .contents {
    width: 60vw;
    height: auto;
    margin-left: auto;
    margin-right: auto;
    border: 0;

    @media (max-width: 768px) {
      width: 90vw;
    }
  }
}
