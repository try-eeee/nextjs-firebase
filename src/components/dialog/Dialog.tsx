"use client";

import styles from "./Dialog.module.scss";
import { forwardRef, RefObject, useCallback } from "react";

interface DialogProps {
  children: React.ReactNode;
  handleOutside: () => void;
}

export function useDialog(ref: RefObject<HTMLDialogElement>) {
  const showModal = useCallback(() => {
    if (ref.current) {
      ref.current.showModal();
    }
  }, [ref]);

  const closeModal = useCallback(() => {
    if (ref.current) {
      ref.current.close();
    }
  }, [ref]);

  return { ref, showModal, closeModal };
}

export const Dialog = forwardRef<HTMLDialogElement, DialogProps>(
  function DialogBase(props, ref) {
    const { children, handleOutside } = props;
    return (
      <dialog ref={ref} onClick={handleOutside} className={styles.dialog}>
        <div className={styles.contents} onClick={(e) => e.stopPropagation()}>
          {children}
        </div>
      </dialog>
    );
  }
);
