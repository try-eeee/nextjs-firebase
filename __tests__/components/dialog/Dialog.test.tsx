import React from "react";
import { render } from "@testing-library/react";
import { Dialog } from "@/components/dialog/Dialog";

describe("Dialog UnitTest", () => {
  test("Snapshot Test Of Dialog", () => {
    const { asFragment } = render(
      <Dialog handleOutside={() => {}}>
        <>コンテンツ</>
      </Dialog>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
