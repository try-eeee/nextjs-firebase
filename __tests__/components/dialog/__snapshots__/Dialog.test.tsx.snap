// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog UnitTest Snapshot Test Of Dialog 1`] = `
<DocumentFragment>
  <dialog
    class="dialog"
  >
    <div
      class="contents"
    >
      コンテンツ
    </div>
  </dialog>
</DocumentFragment>
`;
